timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/impl/PrivilegeBasedNamespaceLocker.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.impl;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.palantir.atlasdb.timelock.auth.api.Privileges;
+import com.palantir.lock.TimelockNamespace;
+import com.palantir.atlasdb.timelock.auth.api.Client;
+
+public class PrivilegeBasedNamespaceLocker implements NamespaceLocker {
+    private List<Privileges> nonAdminPrivileges;
+
+    PrivilegeBasedNamespaceLocker(Map<Client, Privileges> privileges) {
+        this.nonAdminPrivileges = privileges.values().stream()
+                .filter(privilege -> privilege != Privileges.ADMIN)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean isLocked(TimelockNamespace namespace) {
+        return nonAdminPrivileges.stream().anyMatch(privilege -> privilege.hasPrivilege(namespace));
```

Comment:
don't think this is what you want, if this is *just* checking whether or not the namespace is locked, it shouldn't check whether or not then namespace has privilege

Reply:
ignore me

Reply:
however, this will suck, since we'd have like 80 or so clients on current deployments

Reply:
basically I think you should merge your config PR #4042 in first, and work with what you've got, otherwise you're providing impls that are somewhat pointless and will change immediately.

I would in the constructor or whatever, take in the privilege map, visit each privilege, and extract the static namespaces into a set, and then additionally extract the regexes (and that'd be the only thing you'd have to loop through, but you can slap on a cache there).

Reply:
yeah, I know this one is not optimal for perf, I was thinking of having a CachedAuthenticator to prevent going through this path all the time. Wanted to do it in a separate pr though.